envoy/upstream/load_balancer.h

```diff
@@ -103,6 +105,16 @@ class LoadBalancerContext {
    * and return the corresponding host directly.
    */
   virtual absl::optional<OverrideHost> overrideHostToSelect() const PURE;
+
+  // Invoked when a new orca report is received for this LB context.
+  using OrcaLoadReportCb =
+      std::function<absl::Status(const xds::data::orca::v3::OrcaLoadReport& orca_load_report)>;
+
+  /**
+   * Install a callback to be invoked when ORCA Load report is received for this
+   * LB context.
+   */
+  virtual void setOrcaLoadReportCb(OrcaLoadReportCb callback) PURE;
```

Comment:
nit: rather than a function wrapper, I will prefer to define a new interface like `OrcaLoadReportCallbacks` to avoid possible heap allocation of the function wrapper.

Reply:
Good point! Done.